src/app/clusters/mode-base-server/mode-base-server.h

```diff
@@ -149,19 +112,56 @@ class Instance : public CommandHandlerInterface, public AttributeAccessInterface
      */
     bool IsSupportedMode(uint8_t mode);
 
-    /**
-     * Creates a mode base cluster instance. The Init() function needs to be called for this instance to be registered and
-     * called by the interaction model at the appropriate times.
-     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
-     * @param aClusterId The ID of the ModeBase aliased cluster to be instantiated.
-     * @param aFeature The bitmask value that identifies which features are supported by this instance.
-     */
-    Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeature);
+private:
+    Delegate * mDelegate;
 
-    ~Instance() override;
+    EndpointId mEndpointId{};
+    ClusterId mClusterId{};
+
+    // Attribute data store
+    uint8_t mCurrentMode;
+    DataModel::Nullable<uint8_t> mStartUpMode;
+    DataModel::Nullable<uint8_t> mOnMode;
+    uint32_t mFeature;
 
     template <typename RequestT, typename FuncT>
     void HandleCommand(HandlerContext & handlerContext, FuncT func);
+
+    // CommandHandlerInterface
+    void InvokeCommand(HandlerContext & ctx) override;
+    CHIP_ERROR EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context) override;
+    CHIP_ERROR EnumerateGeneratedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context) override;
+
+    // AttributeAccessInterface
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+
+    /**
+     * Register this ModeBase instance.
```

Comment:
Any way to update these doccomments and below to state more than just what can be read from the method name? If the method name is not obvious that `this` is being used to register, maybe `RegisterThisInstance` and such is a slightly easier to read name.

Reply:
Updated.